### PR TITLE
[flutter_tools] ensure linux doctor validator finishes when pkg-config is not installed

### LINKED ISSUE #103744: Upgrade to Flutter 3.0 on ChromeBook error finding pkg-config
bill@penguin:/$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.0, on Debian GNU/Linux 11 (bullseye)
    5.10.106-15290-g6e9d70c8a9b4, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
[✓] Chrome - develop for the web
[☠] Linux toolchain - develop for Linux desktop (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message
      below is not helpful, please let us know about this issue at
      https://github.com/flutter/flutter/issues.
    ✗ ProcessException: Failed to find "pkg-config" in the search path.
        Command: pkg-config 
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.67.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

my chromebook running debian 11.3

### PR DESCRIPTION
**Pull request by christopherfujino**
A follow-up fix to https://github.com/flutter/flutter/pull/100159

Even though we were catching the initial exception when we called `pkg-config --version`, we then followed to call `pkg-config` again to look up gtk libraries, so the exception bubbled up again into the doctor. Instead, only validate the gtk libraries if we know pkg-config is installed.

Fixes https://github.com/flutter/flutter/issues/103744

### CODE REVIEW

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Review by stuartmorgan-g — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/linux/linux_doctor.dart [hunk 1]
```diff
@@ -127,6 +127,8 @@ class LinuxDoctorValidator extends DoctorValidator {
       final _VersionInfo? version = installedVersions[kPkgConfigBinary];
       if (version == null || version.number == null) {
         messages.add(ValidationMessage.error(_userMessages.pkgConfigMissing));
+        // exit early because we cannot validate libraries without pkg-config
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/linux/linux_doctor.dart:130]: nit: Start comment with a capital letter, and end with a period.

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/lib/src/linux/linux_doctor.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/linux/linux_doctor.dart]: done

**Review by cbracken — APPROVED** [review]
🎉

**Pull request merged**